package-res/reportviewer/reportviewer-prompt.js

```diff
@@ -73,10 +75,140 @@ define([
+          if("parameters" == parametersGroup.group) {
+            for (var parameter in parametersGroup.parameters) {
+              if (currentParameterSet.hasOwnProperty(parameter)) {
+                if(JSON.stringify(parametersGroup.parameters[parameter]) !== JSON.stringify(currentParameterSet[parameter])) {
+                  // add to changed
+                  changedParameters.push(parameter);
+                }
+              } else {
+                // add to changed
+                changedParameters.push(parameter);
+              }
+            }
+          }
+        });
+
+        for (var parameter in currentParameterSet) {
+          var foundInSystemParams = false;
+          var foundInParams = false;
+          $.each(oldParameterSet, function (i, parametersGroup) {
+            if("parameters" == parametersGroup.group) {
+              if (parametersGroup.parameters.hasOwnProperty(parameter)) {
+                foundInParams = true;
+              }
+            } else {
+              if (parametersGroup.parameters.hasOwnProperty(parameter)) {
+                foundInSystemParams = true;
+              }
+            }
+          });
+          if(!foundInParams && !foundInSystemParams) {
+            // add to changed
+            changedParameters.push(parameter);
+          }
+        }
+
+        return changedParameters;
+      },
+
+      extractParameterValues: function(paramDefn) {
+        var selectedParameters = [];
+        $.each(paramDefn.parameterGroups, function (i, group) {
+          var parameters = group.parameters;
+          var extractedParameters = {};
+          for(var i=0; i<parameters.length; i++) {
+            if(parameters[i].multiSelect && parameters[i].getSelectedValuesValue().length > 0) {
+              extractedParameters[parameters[i].name] = parameters[i].getSelectedValuesValue();
+            } else {
+              if(parameters[i].getSelectedValuesValue().length > 0) {
+                extractedParameters[parameters[i].name] = parameters[i].getSelectedValuesValue()[0];  
+              }
+            }
+          }
+
+          var selectedParametersGroup = {
+            group : group.name,
+            parameters : extractedParameters
+          };
+
+          selectedParameters.push(selectedParametersGroup);
+        });
+
+        return selectedParameters;
+      },
+
+      findChangedParameters: function() {
+        var currentParameterSet = this.api.operation.getParameterValues();
+
+        // compare currentParameterSet with oldParmaeterSet. Return an array of changed parameters. (More than one can change if we use the API).
+        var changedParameters = this.compareParameters(this._oldParameterSet, currentParameterSet);        
+        return changedParameters;
+      },
+
+      canSkipParameterChange: function(names) {
+        var outputTarget;
+        $.each(this._oldParameterSet, function (i, parametersGroup) {
```

Comment:
This test is to restrictive. The parameter-UI in non-async mode will send the same parameter set twice, once for renderMode=XML and one for renderMode=PARAMETER. 

Implement a general rule that states that if any of the "system" parameters or the "output-target" changes, then you don't skip this request.